elastic/eui
# [EuiInlineEdit] DOM cleanup, a11y improvements, and fix breaking `onChange` bug

## Summary

This PR originally started as just screen reader improvements to the `isLoading` state and then I ended up doing a bunch of other misc cleanup and fixes 😅 I recommend checking individual commit messages for details and reasonings behind changes:

- DOM cleanup: 894da9a0a65065913f2ef8027cdab8f615e92915
- `isLoading` screen reader UX improvement:
  - d1b75f98f345af03389e787ff39368b89883517d
  - 76e04a49d4e416439bf37d2ff3839c7a484cb81c
- Read mode screen reader UX improvement: ac19b54e97354398ccbcc849e6cad54ea9f6c54d
- `onChange` bugfix: a5001a7a371a67591fb0cb975c47140ddf1004ca
- `autoFocus` fix: a236a56177674bbffdcf44678987bb8ff9f59d1f

## QA

- Go to http://localhost:8030/#/display/inline-edit#validating-edited-text
- Start VoiceOver
- Tab to the inline edit read mode button (for the Validation example specifically)
- [x] Confirm that the button accurately describes what will happen on click
- Press enter to toggle edit mode
- [x] Confirm your screen reader does not read out "Loaded" 2x (compare against [staging](https://eui.elastic.co/pr_6743/#/display/inline-edit))
- Tab to and press enter on the save button.
- [x] Confirm that your screen reader reads out a "Loading" status (validation example only)

### General checklist


~- [ ] Added **[documentation](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/documenting)**~ Skipped this for the new EuiSkeleton components - the API is fairly edge case, and I think the props docs sufficiently covers usage
